packages/demo-react-admin/src/App.tsx

```diff
@@ -32,69 +30,67 @@ const App = () => {
-            <Admin
-                authProvider={authProvider}
-                dataProvider={dataProvider}
-                i18nProvider={i18nProvider}
-                title="Example Admin"
-                layout={Layout}
-                loginPage={LoginPage}
-            >
-                {permissions => (
-                    <>
-                        <CustomRoutes noLayout>
-                            <Route
-                                path="/custom"
-                                element={
-                                    <CustomRouteNoLayout title="Posts from /custom" />
-                                }
-                            />
-                        </CustomRoutes>
-                        <Resource name="posts" {...posts} />
-                        <Resource name="comments" {...comments} />
-                        <Resource name="tags" {...tags} />
-                        {permissions ? (
-                            <>
-                                {permissions.includes('admin') ? (
-                                    <Resource name="users" {...users} />
-                                ) : null}
-                                <CustomRoutes noLayout>
-                                    <Route
-                                        path="/custom1"
-                                        element={
-                                            <OneTapButton>
-                                                <CustomRouteNoLayout title="Posts from /custom1" />
-                                            </OneTapButton>
-                                        }
-                                    />
-                                </CustomRoutes>
-                                <CustomRoutes>
-                                    <Route
-                                        path="/custom2"
-                                        element={
-                                            <CustomRouteLayout title="Posts from /custom2">
-                                                <OneTapButton />
-                                            </CustomRouteLayout>
-                                        }
-                                    />
-                                </CustomRoutes>
-                            </>
-                        ) : null}
-                        <CustomRoutes>
-                            <Route
-                                path="/custom3"
-                                element={
-                                    <CustomRouteLayout title="Posts from /custom3">
-                                        <OneTapButton />
-                                    </CustomRouteLayout>
-                                }
-                            />
-                        </CustomRoutes>
-                    </>
-                )}
-            </Admin>
-        </GoogleAuthContextProvider>
+        <Admin
+            authProvider={googleAuthProvider()}
```

Comment:
This is worse, please extract all providers outside the component in all examples.